patch-package
--- a/node_modules/react-native-keychain/android/build.gradle
+++ b/node_modules/react-native-keychain/android/build.gradle
@@ -16,7 +16,6 @@ def safeExtGet(prop, fallback) {
 
 android {
     compileSdkVersion safeExtGet('compileSdkVersion', 26)
-    buildToolsVersion safeExtGet('buildToolsVersion', '26.0.3')
 
   defaultConfig {
     minSdkVersion safeExtGet('minSdkVersion', 16)
@@ -34,6 +33,6 @@ repositories {
 }
 
 dependencies {
-  compile 'com.facebook.react:react-native:+'
-  compile 'com.facebook.conceal:conceal:1.1.3@aar'
+  implementation 'com.facebook.react:react-native:+'
+  implementation 'com.facebook.conceal:conceal:1.1.3@aar'
 }
